src/test/unit/mcmc/covar_adaptation_test.cpp

```diff
@@ -15,15 +15,26 @@ TEST(McmcCovarAdaptation, learn_covariance) {
   target_covar *= 1e-3 * 5.0 / (n_learn + 5.0);
 
   stan::mcmc::covar_adaptation adapter(n);
-  adapter.set_window_params(50, 0, 0, n_learn, logger);
+  adapter.set_window_params(30, 0, 0, n_learn, logger);
```

Comment:
why reduce to 30?

Reply:
Since n_learn is 10 with 0 init_buffer and 0 term_buffer, the first window is size 10, and then the second is size 20. The new check for us is that at the end of the second window (draw 30) learn_covariance doesn't return true (indicating the stepsize should be updated).